ospf6d/ospf6_neighbor.c

```diff
@@ -1035,34 +1024,62 @@ DEFUN (show_ipv6_ospf6_neighbor,
 				json, JSON_C_TO_STRING_PRETTY));
 		json_object_free(json);
 	}
-	return CMD_SUCCESS;
 }
 
+DEFUN(show_ipv6_ospf6_neighbor, show_ipv6_ospf6_neighbor_cmd,
+      "show ipv6 ospf6 [vrf <NAME|all>] neighbor [<detail|drchoice>] [json]",
+      SHOW_STR IP6_STR OSPF6_STR VRF_CMD_HELP_STR
+      "All VRFs\n"
+      "Neighbor list\n"
+      "Display details\n"
+      "Display DR choices\n" JSON_STR)
+{
+	int idx_type = 4;
+	int detail_idx = 5;
+	int json_idx = 6;
+	struct ospf6 *ospf6;
+	struct listnode *node;
+	const char *vrf_name = VRF_DEFAULT_NAME;
+	bool all_vrf = false;
+	int idx_vrf = 0;
+
+	OSPF6_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf);
+	if (idx_vrf > 0) {
+		idx_type += 2;
+		detail_idx += 2;
+		json_idx += 2;
+	}
+	if (all_vrf) {
+		for (ALL_LIST_ELEMENTS_RO(om6->ospf6, node, ospf6)) {
+			ospf6_neighbor_show_detail_common(vty, argc, argv,
+							  ospf6, idx_type,
+							  detail_idx, json_idx);
+		}
+		return CMD_SUCCESS;
+	}
+	ospf6 = ospf6_lookup_by_vrf_name(vrf_name);
+	OSPF6_CMD_CHECK_RUNNING(ospf6);
+
+	ospf6_neighbor_show_detail_common(vty, argc, argv, ospf6, idx_type,
```

Comment:
Same comment as before. This call for specific VRFs could also have been accommodated inside the for loop for 'all_vrf'.

Reply:
Resolved.